src/main/java/teammates/common/datatransfer/FeedbackResponseAttributes.java

```diff
@@ -218,5 +225,52 @@ public int compare(FeedbackResponseAttributes fr1, FeedbackResponseAttributes fr
             }
         });
     }
+
+    public boolean isCorrupted() {
+        return corruptedMessage != null;
+    }
     
+    public void setCorruptionMessage(String errorMessage) {
+        corruptedMessage = errorMessage;
+    }
+    
+    public String constructJsonCorruptedMessage(Class<? extends FeedbackResponseDetails> responseDetailsClass) {
+        if (responseDetailsClass == FeedbackTextResponseDetails.class) {
+            return corruptedMessage;
+        }
+        if (responseDetailsClass == FeedbackConstantSumResponseDetails.class) {
+            return "{\"questionType\" : \"CONTSUM\",\"answers\" : [-1] }";
```

Comment:
The right format json data should be passed to responseDetails class, so I construct the json data here. Do you mean the invalid response message should be handled by ResponseDetails/QuestionDetails page?